File: program/print-reverse-of-a-number-by-recursion/print_reverse_of_a_number_by_recursion.php
<?php

function reverseNumber($num)
{
 $n = strlen($num);
 if($n === 1)
   {
    return $num;
   }
 else
   {
   $n--;
   return reverseNumber(substr($num,1, $n)) . substr($num, 0, 1);
   }
}
print_r(reverseNumber(123)."\n");

?>

File: program/print-reverse-of-a-number-by-recursion/print_reverse_of_a_number_by_recursion.r
reverse_number <- function(n) {
  # base case: if the number has only one digit, return it
  if (n < 10) {
    return(n)
  } else {
    # recursive case: extract the last digit of the number and append it to the
    # reversed version of the remaining digits
    last_digit <- n %% 10
    remaining_digits <- n %/% 10
    return(last_digit * 10^(nchar(as.character(remaining_digits))) + reverse_number(remaining_digits))
  }
}

# example usage
num <- 12345
cat("Original number:", num, "\n")
cat("Reversed number:", reverse_number(num), "\n")
